geocity/apps/accounts/users.py

```diff
@@ -51,10 +51,19 @@ def get_integrator_permissions():
     )
 
 
-def get_users_list_for_integrator_admin(user):
+def get_users_list_for_integrator_admin(user, remove_anonymous=False):
     # Integrators can only view users for restricted email domains.
     if user.is_superuser:
-        return User.objects.all()
+        qs = User.objects.all()
+
+        # Used to remove anonymous users from the list
+        anonymous_users = []
+        for user in qs:
+            if remove_anonymous and user.userprofile.is_anonymous:
```

Comment:
Move the `remove_anonymous` if test above the `for `loop in order to avoid looping when not needed ?